File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-doc-5-4/PartyPortfolioName.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class PartyPortfolioName : ISerialized
    {
        public PartyPortfolioName(XmlNode xmlNode)
        {
            XmlNode partyReferenceNode = xmlNode.SelectSingleNode("partyReference");
            
            if (partyReferenceNode != null)
            {
                if (partyReferenceNode.Attributes["href"] != null || partyReferenceNode.Attributes["id"] != null) 
                {
                    if (partyReferenceNode.Attributes["id"] != null) 
                    {
                        partyReferenceIDRef_ = partyReferenceNode.Attributes["id"].Value;
                        PartyReference ob = new PartyReference(partyReferenceNode);
                        IDManager.SetID(partyReferenceIDRef_, ob);
                    }
                    else if (partyReferenceNode.Attributes["href"] != null)
                    {
                        partyReferenceIDRef_ = partyReferenceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        partyReference_ = new PartyReference(partyReferenceNode);
                    }
                }
                else
                {
                    partyReference_ = new PartyReference(partyReferenceNode);
                }
            }
            
        
            XmlNodeList portfolioNameNodeList = xmlNode.SelectNodes("portfolioName");
            
            if (portfolioNameNodeList != null)
            {
                this.portfolioName_ = new List<PortfolioName>();
                foreach (XmlNode item in portfolioNameNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            portfolioNameIDRef_ = item.Attributes["id"].Value;
                            portfolioName_.Add(new PortfolioName(item));
                            IDManager.SetID(portfolioNameIDRef_, portfolioName_[portfolioName_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            portfolioNameIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        portfolioName_.Add(new PortfolioName(item));
                        }
                    }
                    else
                    {
                        portfolioName_.Add(new PortfolioName(item));
                    }
                }
            }
            
        
        }
        
    
        #region PartyReference_
        private PartyReference partyReference_;
        public PartyReference PartyReference_
        {
            get
            {
                if (this.partyReference_ != null)
                {
                    return this.partyReference_; 
                }
                else if (this.partyReferenceIDRef_ != null)
                {
                    partyReference_ = IDManager.getID(partyReferenceIDRef_) as PartyReference;
                    return this.partyReference_; 
                }
                else
                {
                      return this.partyReference_; 
                }
            }
            set
            {
                if (this.partyReference_ != value)
                {
                    this.partyReference_ = value;
                }
            }
        }
        #endregion
        
        public string partyReferenceIDRef_ { get; set; }
        #region PortfolioName_
        private List<PortfolioName> portfolioName_;
        public List<PortfolioName> PortfolioName_
        {
            get
            {
                if (this.portfolioName_ != null)
                {
                    return this.portfolioName_; 
                }
                else if (this.portfolioNameIDRef_ != null)
                {
                    return this.portfolioName_; 
                }
                else
                {
                      return this.portfolioName_; 
                }
            }
            set
            {
                if (this.portfolioName_ != value)
                {
                    this.portfolioName_ = value;
                }
            }
        }
        #endregion
        
        public string portfolioNameIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-doc-5-4/PartyPortfolioName.cpp
// PartyPortfolioName.cpp 
#include "PartyPortfolioName.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

PartyPortfolioName::PartyPortfolioName(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //partyReferenceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* partyReferenceNode = xmlNode->FirstChildElement("partyReference");

   if(partyReferenceNode){partyReferenceIsNull_ = false;}
   else{partyReferenceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- partyReferenceNode , address : " << partyReferenceNode << std::endl;
   #endif
   if(partyReferenceNode)
   {
      if(partyReferenceNode->Attribute("href") || partyReferenceNode->Attribute("id"))
      {
          if(partyReferenceNode->Attribute("id"))
          {
             partyReferenceIDRef_ = partyReferenceNode->Attribute("id");
             partyReference_ = boost::shared_ptr<PartyReference>(new PartyReference(partyReferenceNode));
             partyReference_->setID(partyReferenceIDRef_);
             IDManager::instance().setID(partyReferenceIDRef_,partyReference_);
          }
          else if(partyReferenceNode->Attribute("href")) { partyReferenceIDRef_ = partyReferenceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { partyReference_ = boost::shared_ptr<PartyReference>(new PartyReference(partyReferenceNode));}
   }

   //portfolioNameNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* portfolioNameNode = xmlNode->FirstChildElement("portfolioName");

   if(portfolioNameNode){portfolioNameIsNull_ = false;}
   else{portfolioNameIsNull_ = true;}

   if(portfolioNameNode)
   {
      for(portfolioNameNode; portfolioNameNode; portfolioNameNode = portfolioNameNode->NextSiblingElement("portfolioName")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- portfolioNameNode , address : " << portfolioNameNode << std::endl;
          #endif
          if(portfolioNameNode->Attribute("href") || portfolioNameNode->Attribute("id"))
          {
              if(portfolioNameNode->Attribute("id"))
              {
                  portfolioNameIDRef_ = portfolioNameNode->Attribute("id");
                  portfolioName_.push_back(boost::shared_ptr<PortfolioName>(new PortfolioName(portfolioNameNode)));
                  portfolioName_.back()->setID(portfolioNameIDRef_);
                  IDManager::instance().setID(portfolioNameIDRef_, portfolioName_.back());
              }
              else if(portfolioNameNode->Attribute("href")) { portfolioNameIDRef_ = portfolioNameNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { portfolioName_.push_back(boost::shared_ptr<PortfolioName>(new PortfolioName(portfolioNameNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- portfolioNameNode , address : " << portfolioNameNode << std::endl;
       #endif
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<PartyReference> PartyPortfolioName::getPartyReference()
{
   if(!this->partyReferenceIsNull_){
        if(partyReferenceIDRef_ != ""){
             return boost::shared_static_cast<PartyReference>(IDManager::instance().getID(partyReferenceIDRef_));
        }else{
             return this->partyReference_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<PartyReference>();
   }
}
std::vector<boost::shared_ptr<PortfolioName>> PartyPortfolioName::getPortfolioName()
{
   if(!this->portfolioNameIsNull_){
        if(portfolioNameIDRef_ != ""){
             return this->portfolioName_;
        }else{
             return this->portfolioName_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<PortfolioName>>();
   }
}
}

